src/tests/ftest/util/pool_create_all_base.py

```diff
@@ -18,58 +22,321 @@ def __init__(self, *args, **kwargs):
+            dmg_scm_size,
+            dmg_nvme_size)
+        self.pool[pool_idx].scm_size.update(dmg_scm_size, "pool[{}].scm_size", pool_idx)
+        if dmg_nvme_size > 0:
+            self.pool[pool_idx].nvme_size.update(dmg_nvme_size, "pool[{}].nvme_size", pool_idx)
+        if ranks is not None:
+            self.pool[pool_idx].target_list.update(ranks, "pool[{}].target_list".format(pool_idx))
+        self.pool[pool_idx].create()
+        self.pool[pool_idx].destroy()
+        pool_idx += 1
+
+        self.log.info(
+            "Creating a pool with SCM oversubscription: scm_size=%d nvme_size=%d",
+            dmg_scm_size + scm_delta_bytes,
+            dmg_nvme_size)
+        self.pool[pool_idx].scm_size.update(
+            dmg_scm_size + scm_delta_bytes,
+            "pool[{}].scm_size".format(pool_idx))
+        if dmg_nvme_size > 0:
+            self.pool[pool_idx].nvme_size.update(
+                dmg_nvme_size,
+                "pool[{}].nvme_size".format(pool_idx))
+        if ranks is not None:
+            self.pool[pool_idx].target_list.update(ranks, "pool[{}].target_list".format(pool_idx))
+        error_msg = r"Pool should not be created: SCM oversubscription"
+        with self.assertRaises(TestFail, msg=error_msg) as context_manager:
+            self.pool[pool_idx].create()
+        self.assertIn(
+            "DER_NOSPACE",
+            str(context_manager.exception),
+            "Pool creation failed with invalid error message")
+        pool_idx += 1
+
+        if dmg_nvme_size > 0:
+            self.log.info(
+                "Creating a pool with NVME oversubscription: scm_size=%d, nvme_size=%d",
+                dmg_scm_size,
+                dmg_nvme_size + nvme_delta_bytes)
+            self.pool[pool_idx].scm_size.update(dmg_scm_size, "pool[{}].scm_size".format(pool_idx))
+            self.pool[pool_idx].nvme_size.update(
+                dmg_nvme_size + nvme_delta_bytes,
+                "pool[{}].nvme_size".format(pool_idx))
+            if ranks is not None:
+                self.pool[pool_idx].target_list.update(
+                    ranks,
+                    "pool[{}].target_list".format(pool_idx))
+            error_msg = r"Pool should not be created: NVME oversubscription"
+            with self.assertRaises(TestFail, msg=error_msg) as context_manager:
+                self.pool[pool_idx].create()
+            self.assertIn(
+                "DER_NOSPACE",
+                str(context_manager.exception),
+                "Pool creation failed with invalid error message")
+            pool_idx += 1
+
+        self.log.info(
+            "Creating a pool with 100%% of the available storage: scm_size=%d, nvme_size=%d",
+            dmg_scm_size,
+            dmg_nvme_size)
+        self.pool[pool_idx].scm_size.update(dmg_scm_size, "pool[{}].scm_size".format(pool_idx))
+        if dmg_nvme_size > 0:
+            self.pool[pool_idx].nvme_size.update(
```

Comment:
In fact, I am creating a first pool with 100% size (at line 100) and then I store the size of the pool dmg_scm_size and dmg_nvme_size variables.
This allows me to create pools (at line 191 and 132), with the exact same size as the first one, using `--scm-size` and `--nvme-size` parameters.
Using the same parameters, I am also able to try to create pools (at lines 151 and 173) with few scm or nvme extra bytes and check that they could not be created (i.e. ENOSPACE error).
By this way, I am able to check that the first pool created was the biggest that we could create.

Reply:
Thanks for the explanation. The comment says, "the pool create with percentage can not be any more properly used.", so I thought the percentage feature was removed. It would have been better if you could include the high-level summary of the changes you made in the functional tests as you made a big change.